xrootd/protocol/protocol.go

```diff
@@ -157,6 +158,91 @@ func (es EntryStat) Mode() os.FileMode {
 	return mode
 }
 
+// MarshalXrd implements xrootd/protocol.Marshaler
+func (o EntryStat) MarshalXrd(wBuffer *xrdenc.WBuffer) error {
+	if !o.HasStatInfo {
+		return nil
+	}
+
+	idStr := strconv.Itoa(int(o.ID))
+	sizeStr := strconv.Itoa(int(o.EntrySize))
+	flags := 0
+	if o.IsExecutable {
+		flags |= 1
+	}
+	if o.IsDirectory {
+		flags |= 2
+	}
+	if o.IsOther {
+		flags |= 4
+	}
+	if o.IsOffline {
+		flags |= 8
+	}
+	if o.IsReadable {
+		flags |= 16
+	}
+	if o.IsWritable {
+		flags |= 32
+	}
+	if o.IsPoscPending {
+		flags |= 64
+	}
+	flagsStr := strconv.Itoa(flags)
+	mtimeStr := strconv.Itoa(int(o.Mtime))
+
+	wBuffer.WriteBytes([]byte(idStr + " " + sizeStr + " " + flagsStr + " " + mtimeStr))
+	return nil
+}
+
+// UnmarshalXrd implements xrootd/protocol.Unmarshaler
+func (o *EntryStat) UnmarshalXrd(rBuffer *xrdenc.RBuffer) error {
```

Comment:
`dirlist` should probably be modified to reuse this code, right?
instead of writing it like:
```go
o.Entries[i/2].UnmarshalXrd(xrdenc.NewRBuffer([]byte(lines[i+1])))
```

I would do it like so:
```go
var rbuf =xrdenc.NewRBuffer(lines[i+1])
o.Entries[i/2].UnmarshalXrd(rbuf)
```
(we could have `dirlist` work in terms of `[][]byte` instead of converting to `[]string`.)